test/index-test.js

```diff
@@ -0,0 +1,22 @@
+'use strict';
+
+const chai = require('chai');
+const main = require('../index.js');
+const assert = chai.assert;
+
+describe('testing main module', function(){
+  describe('testing main with args', function(){
+    before(() => {
+      this.processArgvTwoBackup = process.argv[2];
+      process.argv[2] = 'gnoevil';
+    });
+
+    after(() => {
+      process.argv[2] = this.processArgvTwoBackup;
+    });
+
+    it('should return Hello friend, your name is: gnoevil', function(){
```

Comment:
Duncan will teach us more in detail about this, but see how your before/after blocks use arrow functions and your it block uses the word "function?" This affects parent scopes. 

The it block has no access to any variables declared in your before/after blocks, therefore, no access to process.argv[2] = 'gnoevil." That's because the explicit use of the word "function" creates a separate scope within that it block so that it can't access outside variables. 

If you ```console.log(process.argv[2])``` after your assert method, you'll see it is undefined.

To solve this, your it block can be changed into an arrow function, i.e.
```
it('should return Hello friend, your name is: gnoevil', () => 
```
It now has access to the process.argv[2] delcared in your before/after blocks.